plugins/content_distribution/lib/kContentDistributionFlowManager.php

```diff
@@ -1617,6 +1623,13 @@ public static function onEntryDeleted(entry $entry)
 				KalturaLog::log("Entry distribution [" . $entryDistribution->getId() . "] already flagged for deletion");
 				continue;
 			}
+			
+			if (!in_array ($entryDistribution->getStatus(), $validStatus))
+			{
+				$entryDistribution->setStatus(EntryDistributionStatus::DELETED);
+				$entryDistribution->setDirtyStatus(EntryDistributionDirtyStatus::NONE);
+				$entryDistribution->save ();
```

Comment:
In that case you're breaking the current behavior, until now ready entry-distributions went through submitDelete and removed from the remote system. I think you should mark them as deleted only after the submitDelete finished (maybe even if they failed).